### PR TITLE
Wishlist context

### PR DESCRIPTION
**Pull request by Varu98**

### CODE REVIEW

File: src/frontend/Contexts/ProductContext.jsx [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import react from "react";
 import axios from "axios";
 import { createContext, useContext } from "react";
 import { useState, useEffect } from "react";
```
Comment (Chakravarthi0) [inline src/frontend/Contexts/ProductContext.jsx:3]: You can import in one line

File: src/frontend/ProductCard/ProductCard.jsx [hunk 2]
```diff
@@ -1,24 +1,60 @@
...
   const { cartState, cartDispatch } = useCart();
+  const { wishlistState, wishlistDispatch } = useWishlist();
+
+  useEffect(() => {
+    console.log(wishlistState);
+  }, [wishlistState]);
+
   useEffect(() => {
     console.log(cartState);
   }, [cartState]);
```
Comment (Chakravarthi0) [inline src/frontend/ProductCard/ProductCard.jsx:20]: I think you forgot to remove these

File: src/frontend/Reducers/wishlistReducer.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      return {
+        ...wishlistState,
+        wishlistItems: [
+          ...wishlistState.wishlistItems.filter(
+            (product) => product._id !== payload
+          ),
+        ],
+      };
+    default:
+      return [...wishlistState];
```
Comment (Chakravarthi0) [inline src/frontend/Reducers/wishlistReducer.js:24]: You don't have to destructure i guess

File: src/frontend/WishlistPage/WishlistPage.jsx [hunk 4]
```diff
@@ -1,15 +1,104 @@
...
 import "./WishlistPage.css";
 const WishlistPage = () => {
+  const { cartState, cartDispatch } = useCart();
+  const { wishlistState, wishlistDispatch } = useWishlist();
+  const { wishlistItems } = wishlistState;
+
   return (
-    <div>
+    <>
       {" "}
```
Comment (Chakravarthi0) [inline src/frontend/WishlistPage/WishlistPage.jsx:14]: You can remove this

**Review by Chakravarthi0 — APPROVED** [review]
suggestion: on navbar wishlist icon, you can show the number of items in wishlist.
Apart from the suggestions, everything looks good.

image

**Review by preethasri — APPROVED** [review]
good to merge

**Pull request merged**